alpenhorn/acquisition.py

```diff
@@ -63,6 +178,99 @@ class FileType(base_model):
     name = pw.CharField(max_length=64)
     notes = pw.TextField(null=True)
 
+    # This dict is an atribute on the class used to hold the set of registered
+    # handlers. Store as a dictionary for easy lookup of handlers by name.
+    _registered_file_types = {}
+
+    @classmethod
+    def register_type(cls, file_info):
+        """Register a new file type with alpenhorn.
+
+        This creates the entry in the FileType table if it does not already exist.
+
+        Parameters
+        ----------
+        file_info : AcqInfoBase
+            AcqInfoBase describing the type of acquisition.
+        """
+
+        try:
+            cls.get(name=file_info._file_type)
+        except pw.DoesNotExist:
+            log.info("Create AcqType entry for \"%s\"" % file_info._file_type)
+            cls.create(name=file_info._file_type)
+
+        # Add to registry
+        cls._registered_file_types[file_info._file_type] = file_info
+
+    @classmethod
+    def check_registration(cls):
+        """Check that all FileTypes known to the database have a
+        registered handler.
+        """
+
+        # Get the list of types names from the database
+        db_file_types = [row[0] for row in cls.select(cls.name).tuples()]
+
+        # Get the names of all the registered types
+        reg_file_types = cls._registered_file_types.keys()
+
+        # Find any missing types
+        missing_file_types = set(db_file_types) - set(reg_file_types)
+
+        if len(missing_file_types):
+            raise RuntimeError('FileTypes %s have no registered handler.' %
+                               repr(missing_file_types))
+
+    def is_type(self, filename, acq, node):
+        """Check if this file can be handled by this file type.
+
+        Parameters
+        ----------
+        filename : string
+            Name of the file.
+        acq : ArchiveAcq
+            The acquisition the file is in.
+        node : StorageNode
+            The node we are importing from. Needed so we can inspect the actual
+            acquisition.
+
+        Returns
+        -------
+        is_type : boolean
+        """
+        return self.file_info._is_type(filename, path.join(node.root, acq.name))
```

Comment:
Ah, right, I didn't have it straight in my mind when I was thinking about the `set_info` methods. For some reason I thought the node root leaks into the file/archive info tables.